src/stores/__tests__/errorStore.test.ts

```diff
@@ -0,0 +1,83 @@
+import IError from '~/models/IError';
+import httpStatusDescriptionCodeList from '~/codeLists/httpStatusDescriptionCodeList';
+import { ErrorStore } from '../errorStore';
+
+describe('errorStore.addError', () => {
+    it('Adds one error to error store', () => {
+        const errorStore = new ErrorStore;
+        const errorMessage = 'This is a test error';
+
+        errorStore.addError(errorMessage);
+
+        expect(errorStore.errorCount).toEqual(1);
+        expect(errorStore.latestError).toEqual(errorMessage);
+    });
+
+    it('Adds one error to error store, then removes it', () => {
+        const errorStore = new ErrorStore;
+        const errorMessage = 'This is a test error';
+
+        errorStore.addError(errorMessage);
+
+        expect(errorStore.errorCount).toEqual(1);
+
+        const error = errorStore.pop();
+
+        expect(error).toEqual(errorMessage);
+        expect(errorStore.errorCount).toEqual(0);
+    });
+
+    it('Adds one error with exception to error store', () => {
+        // tslint:disable:no-console
+        // Mocks console.error
+        const consoleError = console.error;
+        global.console.error = jest.fn();
+
+        const errorStore = new ErrorStore;
+        const errorMessage = 'This is a test error';
+        const exceptionMessage = 'This is a test exception message';
```

Comment:
Hmm, exception & error... both mean the same. errorMessage1, errorMessage2? I wouldn't use exception word

See my comment @ this card's description:
Also search for ".catch(", we now have catch (ex) and catch (err) --> pick 1 for consistency

There is still one catch(err) ...

I would use error everywhere (addError, errorMessage, catch(err), etc)

Reply:
Error does not mean the same thing as an exception. Error means a generic issue that has come up. We could basically write 'Lost contact with the server', without ever throwing an exception. Exception is a standard for throwing and catching errors.

Reply:
https://stackoverflow.com/questions/16142583/whats-the-difference-between-error-and-exception-in-javascript

says: Behind the scenes, the browsers use the same window.Error constructor. An Exception is an Error instance with a name and message that contain "Exception"

So okay that is an exception if it has name & message fields.

```
        const errorStore = new ErrorStore;
        const errorMessage = 'This is a test error';
        const exception: Error = {
            message: 'This is a test exception message',
            name: 'Test exception',
        };
        errorStore.addError(errorMessage, exception);
```

Reply:
Based on that link, we should maybe use catch (e) because we don't know if its an exception or error